cmake/kconfig.cmake

```diff
@@ -23,19 +23,17 @@ if(OVERLAY_CONFIG)
   string(REPLACE " " ";" OVERLAY_CONFIG_AS_LIST "${OVERLAY_CONFIG}")
 endif()
 
-set(ENV{srctree}            ${ZEPHYR_BASE})
-set(ENV{KERNELVERSION}      ${KERNELVERSION})
-set(ENV{KCONFIG_CONFIG}     ${DOTCONFIG})
-set(ENV{PYTHON_EXECUTABLE} ${PYTHON_EXECUTABLE})
+# Read environmental variables which needs to be set for Kconfig
+# Idealy the variable name and environmental variable name should be the same
+file(STRINGS ${CMAKE_CURRENT_LIST_DIR}/kconfig_env_vars.txt KCONFIG_ENV_VARS_AS_LIST)
+print(KCONFIG_ENV_VARS_AS_LIST)
 
 # Set environment variables so that Kconfig can prune Kconfig source
-# files for other architectures
-set(ENV{ARCH}      ${ARCH})
-set(ENV{BOARD_DIR} ${BOARD_DIR})
-set(ENV{SOC_DIR}   ${SOC_DIR})
-set(ENV{PROJECT_BINARY_DIR} ${PROJECT_BINARY_DIR})
-set(ENV{ARCH_DIR}   ${ARCH_DIR})
-set(ENV{GENERATED_DTS_BOARD_CONF} ${GENERATED_DTS_BOARD_CONF})
+set(srctree ${ZEPHYR_BASE})
+set(KCONFIG_CONFIG ${DOTCONFIG})
+foreach(env_var ${KCONFIG_ENV_VARS_AS_LIST})
+  set(ENV{${env_var}} ${${env_var}})
+endforeach(env_var)
 
 add_custom_target(
   menuconfig
```

Comment:
Could we use this to get rid of the duplication below?

Reply:
I think so I'll try and add it if it works